src/stores/useReportsStore.ts

```diff
@@ -0,0 +1,153 @@
+import {useLocalObservable} from 'mobx-react-lite';
+import authStore from './authStore';
+import {runInAction} from 'mobx';
+import useApiService from '../network/useAPIService';
+import AppStrings from '../utils/AppStrings';
+import Utility from '../utils/Utility';
+import {NativeStackNavigationProp} from '@react-navigation/native-stack';
+import {CustomReportsStackRootParamList} from '../navigation/CustomReportsStack';
+import {useNavigation} from '@react-navigation/native';
+import {writeFile, DownloadDirectoryPath} from 'react-native-fs';
+import XLSX from 'xlsx';
+
+const useReportsStore = () => {
+  const navigation =
+    useNavigation<NativeStackNavigationProp<CustomReportsStackRootParamList>>();
+  const DDP = DownloadDirectoryPath + '/';
+  const output = (str: any) => str;
+  const {request} = useApiService();
+
+  const partnerNameLocation = () => {
+    return authStore.userData.partner_list.map(item => {
```

Comment:
Create a common function as this is used at many places

Reply:
ok anuj